### PR TITLE
Plugins: Migrate `plugincontext`, `adapters` and `pluginsettings` to pkg/services/pluginsintegration package

### PR DESCRIPTION
**Pull request by wbrowne**
Currently we maintain a few different packages for all the different plugin™️  things.

Without going too deep;
- `pkg/plugins/*` - Core plugin management stuff. The default dumping ground for most stuff, but most actual "core" stuff lies under `pkg/plugins/manager/*`
- `pkg/services/pluginsintegration` - Plugins WireSet, client middleware
- `pkg/services/plugindashboards` - Plugin dashboards (dashboard sync, CRUD, etc.)
- `pkg/services/pluginsettings` - App plugins settings (CRUD)

I think it might be beneficial to start keeping Grafana-specific logic relating to plugins under one package such as  `pkg/services/pluginsintegration ` (which seems to follow the normal Grafana service package pattern). This means we have one place for Grafana's plugin needs and therefore we keep `pkg/plugins/*` as core plugin management. This will essentially act as the plugin management library, ideally hermetically sealed from the rest of Grafana. Its longterm home remains to be seen.

**What this PR changes**
Migrates the following to `pkg/services/pluginsintegration` package 
- `pkg/plugins/plugincontext`
- `pkg/plugins/adapters`
- `pkg/services/pluginsettings`

See also a draft that moves `client` and `store` here ➡️  https://github.com/grafana/grafana/pull/64159

### CODE REVIEW

**Review by marefr — COMMENTED** [review]
When adding the pluginsintegration package my idea was to put Grafana specific plugin implementations in that package moving forward, see [doc](https://github.com/grafana/grafana/blob/main/pkg/services/pluginsintegration/doc.go). Open for discussion, but my 2 cents there should either be pkg/services/plugins or either pkg/services/pluginsintegration. Given that we already have a pkg/plugins package I find it more confusing to add pkg/services/plugins.

WDYT?

**Comment by wbrowne** [comment]
@marefr Were you picturing something like:
- `pkg/services/pluginsintegration/pluginsintegration.go` - WireSet
- `pkg/services/pluginsintegration/clientmiddleware/`
- `pkg/services/pluginsintegration/plugincontext/`
- etc.

Yeah I think we're on the same page - the only difference being what the root package name is called (`plugins` vs `pluginsintegration`), which to me as long as it stays only where the WireSet is defined then it should be fine (just because `pluginsintegration` is a pretty long and non-specific package name IMO)

@andresmgot @xnyo - Does it in general make sense and WDYT about naming? Anything to be improved upon?

**Comment by andresmgot** [comment]
as far as I understood, the goal is to make some clear distinction between the packages that are related to core plugins and services used by all the plugins, right? That makes sense to me. Maybe, to make it more explicit, we could rename `pkg/plugins` to `pkg/coreplugins` (or `pkg/core/plugins`). If not, an external contributor (or myself without context), wouldn't know about this distinction, being confusing to have two "plugins" packages.

Regarding, `plugins` or `pluginsintegrations` maybe it's because I don't have much context but `pluginsintegration`, the `integration` suffix doesn't tell me much, but again I am lacking context.

**Comment by marefr** [comment]
> @marefr Were you picturing something like:
> 
> * `pkg/services/pluginsintegration/pluginsintegration.go` - WireSet
> * `pkg/services/pluginsintegration/clientmiddleware/`
> * `pkg/services/pluginsintegration/plugincontext/`
> * etc.

Yep

**Comment by marefr** [comment]
@andresmgot please see https://github.com/grafana/grafana/pull/58132#issue-1434520762 for some historic context/drawings

**Comment by wbrowne** [comment]
> the goal is to make some clear distinction between the packages that are related to core plugins and services used by all the plugins, right?

@andresmgot Not so much. Probably my use of the word "Core" isn't the most appropriate. I'm keeping Core plugins out of the conversation here for the moment - they're fine to stay as they are I think for the time being. This more relates to plugin management (loading of plugins, managing state, provisioning plugins, etc.) versus the Grafana specific implementation of these interfaces. OSS and Enterprise have different ways of handing things that require specific things that only make sense to Grafana, which we don't want necessarily leaking into plugins / the plugin SDK. This is just about finding a way to have a clear line between these things as for the time being they co-exist in the same repo

**Comment by andresmgot** [comment]
okay, I got it now, thanks for the explanations! Then I agree with @marefr, having `pkg/services/pluginsintegration` makes sense for `integrations` of things defined in `pkg/plugins/` (that way it's clear they don't compete for the same `plugins` purpose).

File: pkg/server/wire.go [hunk 1]
```diff
@@ -92,9 +91,9 @@ import (
 	"github.com/grafana/grafana/pkg/services/playlist/playlistimpl"
 	"github.com/grafana/grafana/pkg/services/plugindashboards"
 	plugindashboardsservice "github.com/grafana/grafana/pkg/services/plugindashboards/service"
-	"github.com/grafana/grafana/pkg/services/pluginsettings"
-	pluginSettings "github.com/grafana/grafana/pkg/services/pluginsettings/service"
 	"github.com/grafana/grafana/pkg/services/pluginsintegration"
+	"github.com/grafana/grafana/pkg/services/pluginsintegration/pluginsettings"
+	pluginSettings "github.com/grafana/grafana/pkg/services/pluginsintegration/pluginsettings/service"
```
Comment (marefr) [inline pkg/server/wire.go:96]: These wire providers should maybe be moved into the pluginsintegration package?

**Review by marefr — APPROVED** [review]
LGTM with a comment

**Review by marefr — APPROVED** [review]

**Pull request merged**